### ISSUE TITLE
[enhancement] Add pdf, docx etc support to MediaMessage(now only picture is supported)

### ISSUE THREAD

**Issue by kungfu-software** [Chatie/wechaty#538]
Please help to resolve the issue.
I am actually relay in two chatrooms. The share pdf file wont work.
 - ROM1 somebody send a pdf file:
![screen shot 2017-05-23 at 6 09 33 pm](https://cloud.githubusercontent.com/assets/700550/26344467/0dcdfe24-3fe3-11e7-8896-c5556dd1de08.png)

- ROM2 file didnt receive. In the log i find the follow error:
![screen shot 2017-05-23 at 6 09 47 pm](https://cloud.githubusercontent.com/assets/700550/26344487/1e45109e-3fe3-11e7-8e05-51e7bf7f2ed4.png)

Thanks

**Comment by zixia** [comment]
### Need More Information

Sorry, you did not follow the **ISSUE Template**, and there's not enough information about your problem.

In order to help us to find the problem out, you should provide all the detail information about the issue. Please make sure you are following the ISSUE Template, answer all the question, and the more important: **provide the steps to reproduce the problem**.

cc @mukaiu

**Comment by kungfu-software** [comment]
Will do tomorrow.

**Comment by jinliming2** [comment]
我也遇到这个问题了，尝试发送图片，可以正常发送，但是发送pptx等文件就会显示发送的是一个undefined文件，0字节，代码：
```javascript
/**
 * 发送文件
 * @param {Sayable} to
 * @param {string}  filename
 * @param {Stream}  stream
 * @return {void}
 */
const sendFile = (to, filename, stream) => {
  const savePath = path.join('static', CONFIG.tempPath, filename);
  mkdirp(savePath.substr(0, savePath.lastIndexOf('/')), async err => {
    if (err) {
      return;
    }
    try {
      const fileStream = fs.createWriteStream(savePath);
      fileStream.on('close', async () => {
        await to.say(new MediaMessage(savePath));
        fs.unlinkSync(savePath);
      });
      stream.pipe(fileStream);
    } catch (e) {
      logger.error(`Save file Error: ${e}\n${e.stack}`);
    }
  });
};
```
里面的stream是一个http下载的文件流。
尝试改filename为英文也没用。

**Comment by mukaiu** [comment]
@jinliming2 pptx is not supported. Late support.

**Comment by jinliming2** [comment]
现在是只支持图片吗？
因为我试了一下，只有图片能正常发送，其他的比如txt也会失败。
但是web微信上手动发送是可以的。

**Comment by lijiarui** [comment]
Hi, @kungfu-software  Can you send pdf now?

**Comment by lijiarui** [comment]
Cannot send pdf too,  It can send pic successfully but not pdf file.
get the following sending result:
![image](https://user-images.githubusercontent.com/6419514/28995681-4019a562-7a22-11e7-8690-d01a19395910.png)

log as follows:

```
TaskList: #TALK #3 #SUCCEED #interval: 3 #taskLog: broadcast file
21:02:29 SILL Message constructor() SN:6
21:02:29 VERB Room say(5步获得AWS的6万云服务代金券.pdf, )
TALK Task Done
21:02:29 VERB PuppetWebBridge getBaseRequest()
21:02:29 VERB PuppetWebBridge proxyWechaty(getBaseRequest, )
21:02:29 SILL PuppetWebBridge execute()
21:02:29 SILL PuppetWebBrowser Browser.execute("return typeof WechatyBro === "undefined"")
21:02:29 SILL StateSwitch Browser:target() is open
21:02:29 SILL StateSwitch Browser:current() is open
21:02:29 SILL PuppetWebBridge execute()
21:02:29 SILL PuppetWebBrowser Browser.execute(" const callback = arguments[arguments.length - 1] const isAsync = (typeof  ... ")
21:02:29 SILL StateSwitch Browser:target() is open
21:02:29 SILL StateSwitch Browser:current() is open
21:02:29 VERB PuppetWebBridge getPassticket()
21:02:29 VERB PuppetWebBridge proxyWechaty(getPassticket, )
21:02:29 SILL PuppetWebBridge execute()
21:02:29 SILL PuppetWebBrowser Browser.execute("return typeof WechatyBro === "undefined"")
21:02:29 SILL StateSwitch Browser:target() is open
21:02:29 SILL StateSwitch Browser:current() is open
21:02:29 SILL PuppetWebBridge execute()
21:02:29 SILL PuppetWebBrowser Browser.execute(" const callback = arguments[arguments.length - 1] const isAsync = (typeof  ... ")
21:02:29 SILL StateSwitch Browser:target() is open
21:02:29 SILL StateSwitch Browser:current() is open
21:02:29 VERB PuppetWebBridge getUploadMediaUrl()
21:02:29 VERB PuppetWebBridge proxyWechaty(getUploadMediaUrl, )
21:02:29 SILL PuppetWebBridge execute()
21:02:29 SILL PuppetWebBrowser Browser.execute("return typeof WechatyBro === "undefined"")
21:02:29 SILL StateSwitch Browser:target() is open
21:02:29 SILL StateSwitch Browser:current() is open
21:02:29 SILL PuppetWebBridge execute()
21:02:29 SILL PuppetWebBrowser Browser.execute(" const callback = arguments[arguments.length - 1] const isAsync = (typeof  ... ")
21:02:29 SILL StateSwitch Browser:target() is open
21:02:29 SILL StateSwitch Browser:current() is open
21:02:29 VERB PuppetWebBrowserCookie read()
21:02:29 SILL PuppetWebBrowserCookie read() webwx_data_ticket,MM_WX_SOUND_STATE,webwx_auth_ticket,MM_WX_NOTIFY_STATE,mm_lang,wxloadtime,wxpluginkey,pgv_si,wxsid,wxuin,webwxuvid,pgv_pvi
21:02:29 VERB PuppetWebBrowser hostname()
21:02:29 SILL PuppetWebBrowser Browser.execute("return document.domain")
21:02:29 SILL StateSwitch Browser:target() is open
21:02:29 SILL StateSwitch Browser:current() is open
21:02:29 SILL PuppetWebBrowser hostname() got qq.com
21:02:29 VERB PuppetWeb self()
21:02:31 SILL PuppetWebEvent onServerDing(heartbeat@browser)
21:02:31 SILL PuppetWebWatchdog onFeed: 60000, HEARTBEAT[heartbeat@browser]
21:02:31 SILL StateSwitch Puppet:target() is live
21:02:31 SILL StateSwitch Puppet:inprocess() is false
21:02:31 SILL PuppetWebWatchdog clearWatchDogTimer() [45] seconds left
21:02:31 SILL PuppetWebWatchdog setWatchDogTimer(60000, HEARTBEAT:[heartbeat@browser])
21:02:31 SILL PuppetWebWatchdog monitorScan(HEARTBEAT)
21:02:31 SILL PuppetWebWatchdog autoSaveSession()
21:02:31 SILL PuppetWebWatchdog memoryCheck() free: 29 MB, require: 4 MB
21:02:31 SILL PuppetWeb send() destination: @@688884a466d382ba5040e2c0bfbe7fa327ac813f4edb6169b2620431f8a385ee, mediaId: @crypt_351c728f_7f1c3d10846338c49d6439fbd8905cd9e6f2b2af3b04e4e4d5ab428c45bdc809ee7c083f0c5581866c6160b38618077469013b833539b6fb6df8342bf10281537534ec64a4e84e630902eb0702007606)
21:02:31 VERB PuppetWebBridge proxyWechaty(sendMedia, @@688884a466d382ba5040e2c0bfbe7fa327ac813f4edb6169b2620431f8a385ee, @crypt_351c728f_7f1c3d10846338c49d6439fbd8905cd9e6f2b2af3b04e4e4d5ab428c45bdc809ee7c083f0c5581866c6160b38618077469013b833539b6fb6df8342bf10281537534ec64a4e84e630902eb0702007606, 49)
21:02:31 SILL PuppetWebBridge execute()
21:02:31 SILL PuppetWebBrowser Browser.execute("return typeof WechatyBro === "undefined"")
21:02:31 SILL StateSwitch Browser:target() is open
21:02:31 SILL StateSwitch Browser:current() is open
21:02:31 SILL PuppetWebBridge execute()
21:02:31 SILL PuppetWebBrowser Browser.execute(" const callback = arguments[arguments.length - 1] const isAsync = (typeof  ... ")
21:02:31 SILL StateSwitch Browser:target() is open
21:02:31 SILL StateSwitch Browser:current() is open
```

**Comment by lijiarui** [comment]
fix, see #710
